src/util/secp256k1.rs

```diff
@@ -61,7 +61,7 @@ pub struct Secp256k1PrivateKey {
     // serde is broken for secp256k1, so do it ourselves
     #[serde(serialize_with = "secp256k1_privkey_serialize", deserialize_with = "secp256k1_privkey_deserialize")]
     key: LibSecp256k1PrivateKey,
-    compress_public: bool
+    pub compress_public: bool
```

Comment:
Could you instead add a `set_compressed()` method to `Secp256k1PrivateKey`, like there is with `Secp256k1PublicKey`?

Reply:
Addressed in d0de5c9cc